packages/salesforcedx-utils-vscode/src/commands/preconditionCheckers.ts

```diff
@@ -34,3 +34,23 @@ export class SfWorkspaceChecker implements PreconditionChecker {
     return true;
   }
 }
+
+/** Runs multiple precondition checks in sequence */
+export class CompositePreconditionChecker implements PreconditionChecker {
+  public checks: PreconditionChecker[];
+
+  constructor(...checks: PreconditionChecker[]) {
+    this.checks = checks;
+  }
+
+  public async check(): Promise<boolean> {
+    for (const output of this.checks) {
+      const input = await output.check();
+      if (input === false) {
+        return false;
+      }
```

Comment:
Can consolidate into 1 line 🙂

```suggestion
      if (!await output.check()) return false;
```